### DISCUSSION TITLE
Phrase search via Searchkit?

### DISCUSSION THREAD

**Discussion by dkalpakchi** [searchkit/searchkit#1243] [Q&A] [answered]
Hi!

First of all, thank you for this absolutely fantastic project!

I'm looking for a way to implement a phrase search, meaning when you search for "Hello world", I want to get only docs that have exactly this phrase.

I think Algolia's instantsearch provides advancedSyntax handle (https://www.algolia.com/doc/api-reference/api-parameters/advancedSyntax/) for that. I couldn't find where (or whether) this could be turned on in SearchKit.

Any help would be much appreciated!

Many thanks!

**Comment by joemcelroy** [comment, marked as answer]
Great to hear @dkalpakchi! 

Yes so best way is to override the default query via `getQuery` function. You can read more about it here https://www.searchkit.co/docs/guides/customising-query

Sounds like you want the [phase match query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html#type-phrase) clause, so would expect something like

```ts
const results = await apiClient.handleRequest(req.body, {
  getQuery: (query, search_attributes) => {
    return [
      {
       "multi_match" : {
          "query":     query,
          "type":       "phrase",
          "fields":     search_attributes
        }
      },
    ];
  },
});
```    

and you can also wrap the query with a bool so ranks docs with phrase matches but still matches on query terms. 

Hope this helps!

Joe

**Reply by dkalpakchi** [reply]
Hi!

Thank you very much for a swift reply! I've been able to try this and it seems to work just fine. For future reference, if someone wants to do something similar, the whole code looks like this for me:

```js
const sk = new Searchkit({
  // your settings here
})

const search = instantsearch({
  indexName: "your-index-name",
  searchClient: SearchkitInstantsearchClient(sk, {
    getQuery: (query, search_attributes) => {
      return [
        {
          "multi_match" : {
            "query":     query,
            "type":       "phrase",
             "fields":     search_attributes
          }
        },
      ];
    },
  })
});
```
